dep-sandbox/src/main/java/com/wso2telco/services/dep/sandbox/servicefactory/payment/PaymentListTransactionRequestHandler.java

```diff
@@ -0,0 +1,176 @@
+                    "msisdn", msisdn, object);
+
+            String tableName = TableName.NUMBERS.toString().toLowerCase();
+            PaymentListTransactionResponseBean paymentTransaction = new PaymentListTransactionResponseBean();
+            amountTransaction = paymentDAO.getTransactionValue(endUserId, attributeName, tableName, userId);
+
+            List<JsonNode> listNodes = new ArrayList<JsonNode>();
+
+            if (amountTransaction != null && !amountTransaction.isEmpty()) {
+                for (AttributeValues values : amountTransaction) {
+                    JsonParser parser = new JsonParser();
+                    JsonObject jsonObject = parser.parse(values.getValue()).getAsJsonObject();
+                    JsonElement get = jsonObject.get("paymentAmount");
+                    JsonObject asJsonObjectPayment = get.getAsJsonObject();
+                    asJsonObjectPayment.remove("totalAmountCharged");
+                    asJsonObjectPayment.remove("totalAmountRefunded");
+                    asJsonObjectPayment.remove("chargingMetaData");
+                    jsonObject.remove("clientCorrelator");
+                    jsonObject.remove("notifyURL");
+                    jsonObject.remove("originalReferenceCode");
+                    jsonObject.remove("originalServerReferenceCode");
+                    jsonObject.remove("resourceURL");
+                    String jsonInString = null;
+                    jsonInString = jsonObject.toString();
+
+                    JsonNode node = null;
+                    ObjectMapper mapper = new ObjectMapper();
+                    node = mapper.readValue(jsonInString, JsonNode.class);
+                    listNodes.add(node);
+                }
+                paymentTransaction.setAmountTransaction(listNodes);
+            } else {
+                LOG.error("###PAYMENT### Valid Transaction List Not Available for msisdn: " + endUserId);
+                responseWrapper.setHttpStatus(Response.Status.NO_CONTENT);
+                responseWrapper.setHttpStatus(Response.Status.OK);
+                return responseWrapper;
+            }
+            paymentTransaction.setResourceURL(CommonUtil.getResourceUrl(extendedRequestDTO));
+
+            PaymentListTransactionDTO listTransactionDTO = new PaymentListTransactionDTO();
+
+            listTransactionDTO.setPaymentTransactionList(paymentTransaction);
+            responseWrapper.setListPaymentDTO(listTransactionDTO);
+            responseWrapper.setHttpStatus(Response.Status.OK);
+
+        } catch (Exception ex) {
+            LOG.error("###PAYMENT### Error Occurred in PAYMENT Service. ", ex);
+            responseWrapper.setHttpStatus(Response.Status.BAD_REQUEST);
+            responseWrapper
+                    .setRequestError(constructRequestError(SERVICEEXCEPTION, ServiceError.SERVICE_ERROR_OCCURED, null));
+            throw ex;
```

Comment:
throw ex; is this a valid statement? 
you are setting some properties to responseWrapper object and throwing an error. Please revisit the logic